shap/plots/_image.py

```diff
@@ -127,3 +132,381 @@ def image(shap_values, pixel_values=None, labels=None, width=20, aspect=0.2, hsp
     cb.outline.set_visible(False)
     if show:
         pl.show()
+
+
+def image_to_text(shap_values):
+
+    from IPython.core.display import display, HTML
+
+    if len(shap_values.values.shape) == 5:
+        for i in range(shap_values.values.shape[0]):
+            display(HTML("<br/><b>"+ordinal_str(i)+" instance:</b><br/>"))
+            image_to_text(shap_values[i])
+        
+        return
+    
+
+    uuid = ''.join(random.choices(string.ascii_lowercase, k=20))
+    
+    # creating input html tokens
+    
+
+    model_output = shap_values.output_names
+
+    output_text_html = ''
+
+    for i in range(model_output.shape[0]):
+        output_text_html += "<div style='display:inline; text-align:center;'>" \
+                + f"<div id='{uuid}_output_flat_value_label_"+ str(i) +"'" \
+                + "style='display:none;color: #999; padding-top: 0px; font-size:12px;'>" \
+                + "</div>" \
+                + f"<div id='{uuid}_output_flat_token_"+ str(i) +"'" \
+                + "style='display: inline; background:transparent; border-radius: 3px; padding: 0px;cursor: default;cursor: pointer;'" \
+                + f"onmouseover=\"onMouseHoverFlat_{uuid}(this.id)\" " \
+                + f"onmouseout=\"onMouseOutFlat_{uuid}(this.id)\" " \
+                + f"onclick=\"onMouseClickFlat_{uuid}(this.id)\" " \
+                + ">" \
+                + model_output[i].replace("<", "&lt;").replace(">", "&gt;").replace(' ##', '').replace('▁', '') \
+                + " </div>" \
+                + "</div>"
+    
+    # computing gray scale images
+    
+    
+    image_data = shap_values.data
+    image_height = image_data.shape[0]
+    image_width = image_data.shape[1]
+    
+    
+    # computing gray scale image
+    
+    image_data_gray_scale = np.ones((image_height, image_width, 4)) * 255 * 0.5
+    image_data_gray_scale[:,:,0] = np.mean(image_data,axis=2).astype(int)
+    image_data_gray_scale[:,:,1] = image_data_gray_scale[:,:,0]
+    image_data_gray_scale[:,:,2] = image_data_gray_scale[:,:,0]
+        
+    # computing shap color values for every pixel and for every output token
+    
+    shap_values_color_maps = shap_values.values[:,:,0,:]
+    max_val = np.nanpercentile(np.abs(shap_values.values), 99.9)
+    
+    shap_values_color_dict = {}
+    
+    for index in range(model_output.shape[0]):        
+        shap_values_color_dict[f'{uuid}_output_flat_token_{index}'] = (colors.red_transparent_blue(0.5 + 0.5 * shap_values_color_maps[:,:,index]/ max_val) * 255).astype(int).tolist()
+
+    
+    # converting to json to be read in javascript
+    
+    image_data_json = json.dumps(shap_values.data.astype(int).tolist())
+    shap_values_color_dict_json = json.dumps(shap_values_color_dict)
+    image_data_gray_scale_json = json.dumps(image_data_gray_scale.astype(int).tolist())
+    
+    image_viz_html = f"""
+
+        <div id="{uuid}_image_viz" class="{uuid}_image_viz_content">
```

Comment:
why do you use {uuid} on all of the divs?  wondering if that could be refactored out somehow, perhaps moved to a function that appends the {uuid} string to the given parameter string

Reply:
The uuid is so there can be multiple plots in the same Jupiter notebook. Not sure how to make a cleaner prefix here, this might be best.